cg/meta/transfer/external_data.py

```diff
@@ -0,0 +1,211 @@
+        slurm_api.set_dry_run(dry_run=dry_run)
+        sbatch_content: str = slurm_api.generate_sbatch_content(Sbatch.parse_obj(sbatch_info))
+        sbatch_path: Path = log_dir.joinpath(str(ticket_id) + "_rsync_external_data.sh")
+        sbatch_number: int = slurm_api.submit_sbatch(
+            sbatch_content=sbatch_content, sbatch_path=sbatch_path
+        )
+        return sbatch_number
+
+    def transfer_sample_files_from_caesar(self, ticket_id: int, dry_run: bool):
+        """Transfers all samples, related to given ticket, from caesar to hasta."""
+        cases: List[models.Family] = self.status_db.get_cases_from_ticket(ticket_id=ticket_id).all()
+        cust: str = self.status_db.get_customer_id_from_ticket(ticket_id=ticket_id)
+        for case in cases:
+            links = case.links
+            for link in links:
+                lims_sample_id: str = link.sample.internal_id
+                cust_sample_id: str = link.sample.name
+                sbatch_number: int = self.transfer_sample(
+                    ticket_id=ticket_id,
+                    cust=cust,
+                    cust_sample_id=cust_sample_id,
+                    lims_sample_id=lims_sample_id,
+                    dry_run=dry_run,
+                )
+                LOG.info(
+                    "Downloading sample %s from caesar by submitting sbatch job %s"
+                    % (lims_sample_id, sbatch_number)
+                )
+
+    def get_all_fastq(self, sample_folder: Path) -> List[Path]:
+        """Returns a list of all fastq.gz files in given folder."""
+        all_fastqs: List[Path] = []
+        for leaf in sample_folder.iterdir():
+            abs_path: Path = sample_folder.joinpath(leaf)
+            LOG.info("Found file %s inside folder %s" % (str(abs_path), sample_folder))
+            if abs_path.suffixes == [".fastq", ".gz"]:
+                all_fastqs.append(abs_path)
+        return all_fastqs
+
+    def get_all_paths(self, lims_sample_id: str, customer: str) -> List[Path]:
+        """Returns the paths of all fastq files associated to the sample."""
+        fastq_folder: Path = self.create_destination_path(
+            lims_sample_id=lims_sample_id, customer=customer, raw_path=self.hasta_path
+        )
+        all_fastq_in_folder: List[Path] = self.get_all_fastq(sample_folder=fastq_folder)
+        return all_fastq_in_folder
+
+    def create_hk_bundle(
+        self, bundle_name: str, dry_run: bool, data_dict: dict
+    ) -> Optional[Tuple[Bundle, Version]]:
+        """Creates and returns a housekeeper bundle and the new version."""
+        if dry_run:
+            LOG.info(
+                "Would have created bundle %s in housekeeper, but this is dry-run", bundle_name
+            )
+            return
+        bundle_result: Tuple[Bundle, Version] = self.housekeeper_api.add_bundle(
+            bundle_data=data_dict
+        )
+        return bundle_result
+
+    def check_bundle_fastq_files(
```

Comment:
Could this not be a generic HK method to check any files that are liinked in HK (not just fastq)? Small tweaks can make this generic and placed in the hk.api instead of defined here